pkg/auth/handlers/handlers.go

```diff
@@ -66,3 +85,25 @@ func unauthorizedBasicAuth(w http.ResponseWriter, req *http.Request) {
 func unauthorized(w http.ResponseWriter, req *http.Request) {
 	http.Error(w, "Unauthorized", http.StatusUnauthorized)
 }
+
+// compressUsername maps all possible usernames onto a small set of categories
+// of usernames. This is done both to limit the cardinality of the
+// authorized_user_requests metric, and to avoid pushing actual usernames in the
+// metric.
+func compressUsername(username string) string {
+	switch {
+	// Known internal identities.
+	case username == "admin" ||
+		username == "client" ||
+		username == "kube_proxy" ||
+		username == "kubelet" ||
+		username == "system:serviceaccount:kube-system:default":
```

Comment:
Agree it's hacky. I just needed a metric that differentiated between requests from some internal identities vs. end-user identities. I don't want to expose real usernames on the `/metrics` endpoint, but I do want to see how often the `admin`/`client` users are being used compared to real users. Are you thinking this would be useful in a more generic way?